### DISCUSSION TITLE
DJL Build ERROR

### DISCUSSION THREAD

**Discussion by xuhongxin** [deepjavalibrary/djl#2852] [Development]
When i build DJL with commond：./gradlew build, the system tip error:
![image](https://github.com/deepjavalibrary/djl/assets/1044109/c4c11e40-0b5e-4841-8ed9-66eff27a9caf)

`> Task :api:compileJava FAILED
Caching disabled for task ':api:compileJava' because:
  Build cache is disabled
Task ':api:compileJava' is not up-to-date because:
  Task has failed previously.
The input changes require a full rebuild for incremental task ':api:compileJava'.
Full recompilation is required because no incremental change information is available. This is usually caused by clean builds or changing compiler arguments.
Compiling with toolchain '/Library/Java/JavaVirtualMachines/jdk-9.0.4.jdk/Contents/Home'.
Compiling with JDK Java compiler API.
`

and i update the gradle.properties config and add 'org.gradle.caching=false' ；then the error change：
![image](https://github.com/deepjavalibrary/djl/assets/1044109/6b9943a7-385d-42f7-b872-e017a96caf24)

`Caching disabled for task ':api:compileJava' because:
  Gradle does not know how file 'build/classes/java/main/ai/djl/engine' was created (output property 'destinationDirectory'). Task output caching requires exclusive access to output paths to guarantee correctness (i.e. multiple tasks are not allowed to produce output in the same location).
Task ':api:compileJava' is not up-to-date because:
  Task has failed previously.
The input changes require a full rebuild for incremental task ':api:compileJava'.
Full recompilation is required because no incremental change information is available. This is usually caused by clean builds or changing compiler arguments.
Compiling with toolchain '/Library/Java/JavaVirtualMachines/jdk-9.0.4.jdk/Contents/Home'.
Compiling with JDK Java compiler API.
`

it seems like 'org.gradle.caching' can influence the build result, but i don't know how to resovle it.

**Comment by zachgk** [comment]
Try running `./gradlew clean` and then trying `./gradlew build`

**Reply by xuhongxin** [reply]
i have try './gradlew clean' but it still report the error

**Comment by xuhongxin** [comment]
more infomation：
i have try jdk8,jdk9 but still have the error
```

Execution failed for task ':api:compileJava'.
> 指定了 --release 选项, 但提供的 JavaFileManager 不是 StandardJavaFileManager
```

**Comment by xuhongxin** [comment]
![image](https://github.com/deepjavalibrary/djl/assets/1044109/ac88e89a-6ea8-4723-bd22-5e208689bafc)

and my jdk is  1.8.0_271 ; and `./gradlew -version` result is gradle 8.3 ; is it not compatiable？which version of jdk and gradle should i chose？

**Comment by frankfliu** [comment]
See: https://github.com/deepjavalibrary/djl/blob/master/docs/development/setup.md

To build DJL from source, you need JDK 11+. But DJL library can run with JDK 1.8

**Reply by xuhongxin** [reply]
thank u
